cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -475,4 +480,17 @@ private NamespaceId getNamespaceId(byte[] rowKey) {
   public interface DeleteCallback {
     void handle(NamespaceId namespaceId, Location location, String namespacedLogBaseDir);
   }
+
+  private static Location getCompatibleLocationFromValue(LocationFactory locationFactory,
+                                                         byte[] value) throws URISyntaxException {
+    String locationPath = null;
+    URI locationURI;
+    String valueString = Bytes.toString(value);
+    locationURI = new URI(valueString);
+    if (locationURI.getPath() == valueString) {
```

Comment:
comparing strings like this will compare their references, not their values. You have to use .equals() to compare strings. So I guess this method only works because the root location factory is used otherwise you would have seen errors?